ci_pasta_keys_verifier_script.py

```diff
@@ -0,0 +1,24 @@
+import subprocess
+import sys
+import os
+
+# python3 ci_pasta_keys_verifier_script.py https://github.com/lurk-lab/Nova.git d5f5fb5e557b99cb111f2d5693aa34fe30722750
+if __name__ == "__main__":
+    # Configurations
+    nova_repo_url = sys.argv[1]
+    nova_commit_hash = sys.argv[2]
+    target_directory = "verify_cache"
+    nova_directory = "nova"
+
+    # Main logic
+    if os.path.exists(target_directory):
+        subprocess.run(['rm', '-rf', target_directory], check=True)
+    os.mkdir(target_directory)
+    os.mkdir(target_directory + "/" + nova_directory)
+    os.chdir(target_directory)
+    os.system(f"git clone {nova_repo_url} {nova_directory}")
+    os.chdir(nova_directory)
+    os.system(f"git checkout {nova_commit_hash}")
+    os.system(f"cargo +nightly test solidity_compatibility_e2e_pasta --release -- --ignored")
```

Comment:
I would make the test name (`solidity_compatibility_e2e_pasta `) also configurable using additional flag. That would allow us changing the application that generates JSONs for us if necessary (for Grumpkin, for example we will use another test).

Reply:
how do you look to just to rename test solidity_compatibility_e2e ? All other steps seems thr same for generating keys for Grumpkin.

Reply:
For grumpkin, I believe you will need separate test, something like:

```
  // cargo +nightly test test_ivc_nontrivial_with_compression_pasta --release -- --nocapture --ignored
  #[test]
  #[ignore]
  fn solidity_compatibility_e2e_pasta() {
    test_ivc_nontrivial_with_compression_with::<bn256::Point, grumpkin::Point>(true);
  }
```
at Nova. So I don't know how to reuse single test for both `pasta` / `grumpkin` JSONs generating.

And also I don't have 100% sure that I used `test_ivc_nontrivial_with_compression` for Grumpkin JSONs initially. We will need to detect the correct commit and test name similarly as we did for pasta (by comparing md5 hashes)

Reply:
how do you resolve this in branch were both _pasta_ / _grumpkin_ are located ?

Reply:
I think, we can add one more commit with mentioned test for grumpkin on top of https://github.com/lurk-lab/Nova/commit/ea4f75c225cb29f523780858ec84f1ff51c229bc. Then, using new commit hash, your script will be able to run two distinct tests and generate appropriate JSONs for each necessary branch in solidity-verifier. Does that make sense?